tests/ragger/test_blind_sign.py

```diff
@@ -47,21 +50,91 @@ def test_blind_sign(firmware: Firmware,
         "data": data,
         "chainId": 1
     }
-    with pytest.raises(ExceptionRAPDU) as e:
-        with app_client.sign(BIP32_PATH, tx_params):
+
+
+# Token approval, would require loading the "internal plugin" &
+# providing the token metadata from the CAL
+def test_blind_sign(firmware: Firmware,
+                    backend: BackendInterface,
+                    navigator: Navigator,
+                    default_screenshot_path: Path,
+                    test_name: str,
+                    sign: bool):
+    global DEVICE_ADDR
+    app_client = EthAppClient(backend)
+
+    if DEVICE_ADDR is None:
+        with app_client.get_public_addr(bip32_path=BIP32_PATH, display=False):
             pass
-    assert e.value.status == StatusWord.INVALID_DATA
+        _, DEVICE_ADDR, _ = ResponseParser.pk_addr(app_client.response().data)
+
+    tx_params = common_tx_params()
+    try:
+        with app_client.sign(BIP32_PATH, tx_params):
+            if sign:
+                test_name += "_signed"
+            else:
+                test_name += "_rejected"
+
+            moves = []
+            if firmware.device.startswith("nano"):
+                if firmware.device == "nanos":
```

Comment:
Also here: `if firmware == Firmware.NANOS`

Reply:
Indeed it could be improved like this, better for a future PR. :+1: